src/VkBootstrap.cpp

```diff
@@ -1648,8 +1648,8 @@ VkSurfaceFormatKHR find_surface_format(VkPhysicalDevice phys_device,
 		}
 	}
 
-	// use the first available one if any desired formats aren't found
-	return available_formats[0];
```

Comment:
I have a feeling that this will result in breaking user code. The fact that the test code had to be modified by adding VK_FORMAT_B8G8R8A8_UNORM makes me leery about this change. Now, that isn't to say the change to return an error instead of the first format isn't good, as it definitely is more consistent & logical. The issue is making sure that the 'person who just wants a swapchain with whatever format is available' is still supported.

Reply:
Ok, I have definitely misunderstood the intent of the "desired" formats. I though here it was returning the first format the user desired, but it is just one format a user requested. It explains a lot why the test was the way it was :)
I will definitely revert this, as changing this behavior was definitely not correct !

Reply:
Well, now that I look at the code, I realize that there isn't a way to specify 'required' formats, which seems like a hole. 

The original design is a 'use the first format the driver gives us *unless* the user asks for something else and it is supported. 
It probably should be amended to allow specifying a list 'required' formats, where the first supported format in the list is chosen, but if no formats are chosen, and error is returned. This could be relatively seamlessly added by having an error be returned if the required list is not empty. So the existing behavior is maintained (if you specify nothing you get the first format), if you specify a desired format, you get it if supported, otherwise the first in the list, and if you speciy a required format, you get it if supported or an error.

Don't feel like you need to implement this, I'm more spitballing ideas than listing asking for changes.

Reply:
Yes, I though at first "desired" meant "required" for some reason, hence my first change.
I specifically left the two separate functions for simple refactoring later (reusing the first function with the error code should be sufficient to check required formats).
Another PR will be best suited for this, but it is certainly a good feature to add !